### PR TITLE
Update Azure PostgreSQL flexible server defaults

### PR DESCRIPTION
**Pull request by eerhardt**
Use the same defaults as the portal does for "Development" mode. This makes the default configuration cheaper for people who are developing/testing. It also uses the latest version (16) instead of 12 by default. Here's the current defaults in the portal:

image

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2391)

### CODE REVIEW

**Review by mitchdenny — APPROVED** [review]
LGTM.

**Review by davidfowl — APPROVED** [review]

File: src/Aspire.Hosting.Azure/Bicep/postgres.bicep [hunk 1]
```diff
@@ -5,12 +5,12 @@ param keyVaultName string
...
-param skuSizeGB int = 128
-param dbInstanceType string = 'Standard_D4ds_v4'
-param haMode string = 'ZoneRedundant'
+param serverEdition string = 'Burstable'
+param skuSizeGB int = 32
+param dbInstanceType string = 'Standard_B1ms'
+param haMode string = 'Disabled'
 param availabilityZone string = '1'
-param version string = '12'
+param version string = '16'
```
Comment (vhvb1989) [inline src/Aspire.Hosting.Azure/Bicep/postgres.bicep:13]: getting this error: 

```
Deployment Error Details:
ParameterOutOfRange: The value of the 'Version' should be in: [13,14,15]. Verify that the specified parameter value is correct.
```

**Comment by vhvb1989** [comment]
The version might depend on the region...

**Comment by DamianEdwards** [comment]
@vhvb1989 docs seem to indicate it should be available in all Azure regions: https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-supported-versions#postgresql-version-16

**Comment by eerhardt** [comment]
What region were you using?

**Comment by vhvb1989** [comment]
I tried eastus2 and australiaeast.
But, I guess my issue is that I had an already created Server using the previous version (15), from before this PR.
Now when I try to run `azd up`, it is trying to update the server... and it complains about the version and then about the disk size :)

```
UpsertServerManagementOperationDowngradeDiskSizeUnsupported: Requested data Disk size '32768' cannot be less than current size '131072'. Please make sure to request correct disk size. Refer to https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-compute-storage for more details.
```

Guess I need to delete the existing one and use a new one.

**Pull request merged**